app/views/users/show.html.erb

```diff
@@ -1,5 +1,12 @@
 <section class="user-section">
-  <h2><%= "Name: #{@user.name}" %></h2>
+  <% if @user.id == current_user.id %>
+    <h2><%= "Welcome #{@user.name}!" %></h2>
+  <% else %>
+    <div class="button-parent">
+      <h2><%= "Name: #{@user.name}" %></h2>
+      <%= render 'friendship_button', user: @user %>    
+  </div>
+  <% end %>
```

Comment:
- A user shouldn't be able to see another user's post unless they are friends. Right now that works great in the Timeline menu, but when you click on a user's profile link, is not working. Please fix that.